apis/run/v1alpha3/tanzukubernetesrelease_types.go

```diff
@@ -40,12 +40,25 @@ type TanzuKubernetesReleaseSpec struct {
 	Kubernetes KubernetesSpec `json:"kubernetes"`
 
 	// OSImages lists references to all OSImage objects shipped with this TKR.
-	OSImages []corev1.LocalObjectReference `json:"osImages,omitempty"`
+	OSImages []OSImageReference `json:"osImages,omitempty"`
 
 	// BootstrapPackages lists references to all bootstrap packages shipped with this TKR.
 	BootstrapPackages []corev1.LocalObjectReference `json:"bootstrapPackages,omitempty"`
 }
 
+// OSImageReference specifies the details about the osimage reference in the TanzuKubernetesRelease objects.
+type OSImageReference struct {
+	corev1.LocalObjectReference `json:",inline"`
+
+	// The URL of public image endpoint where the image will be imported from
+	// +optional
+	ImageURL *string `json:"image_url,omitempty"`
```

Comment:
**nit**: json field should be named `imageURL`: in line with generally accepted style in Kubernetes APIs.